### ISSUE TITLE
Pie Chart is not shown in the browser

### ISSUE THREAD

**Issue by alstat** [hmcuesta/PDA_Book#1]
Hello sir, I am reading your newly published book. And tried some codes in Chapter 3 on data visualization using D3. I followed the instruction in constructing the D3 Pie chart, but when I run it in my local server, nothing is shown except the header I made. I am new to JavaScript, so I wasn't able to fix this. And by the way, I was not successful in plotting the Bar chart too, I tried to use your codes in this repository and renamed sumPokemon.csv to pokemonByType.csv, but still nothing happens. I was hoping if you could help me with this, so that I can proceed with the other charts. Thanks, below is the code.

``` html
<!DOCTYPE html>
<html>
<head>My first D3 Pie Chart</head>
<style>

body {
  font: 16px ARIAL;
}
</style>
<body>
<script src="d3.v3.min.js"></script>
<script>
var w = 1160,
    h = 700,
    radius = Math.min(w, h) / 2;

var color = d3.scale.ordinal()
    .range(["#04B486", "#F2F2F2", "F5F6CE", "#00BFFF"]);

var arc = d3.svg.arc()
    .outerRadius(radius - 10)
    .innerRadius(0);

var pie = d3.layout.pie()
    .sort(null)
    .value(function(d){return d.amount;});

var svg = d3.select("body").append("svg")
    .attr("width", w)
    .attr("height", h)
    .append("g")
    .attr("transform", "translate("+w/2+","+h/2+")");

d3.csv("sumPokemon.csv", function(error, data){
data.forEach(function(d){
d.amount = +d.amount;
});

var g = svg.selectAll(".arc")
    .data(pie(data))
    .enter().append("g")
    .attr("class", "arc");

g.append("path")
    .attr("d", arc)
    .style("fill", function(d){return color(d.data.type);});
g.append("text")
    .attr("transform", function(d){return "translate("+ arc.centroid(d)+")";})
    .attr("dy", ".60em")
    .style("text-anchor", "middle")
    .text(function(d){return d.data.type;});
});
</script>
</body>
</html>
```

**Comment by hmcuesta** [comment]
Hi alstat,

The code looks good and is working in my PC. This example requires a web server like apache or python (python –m http.server 8000). Your code also requires the d3.v3.min.js file in the same folder (You can download it from here http://d3js.org/ ). Finally the file pokemonByType.csv is already in the repo, please refer to the link below.

https://github.com/hmcuesta/PDA_Book/blob/master/Chapter3/pokemonByType.csv

Thanks for reading the book, if I can help you in anything else don't hesitate to ask.

Best Regards

Hector Cuesta

**Comment by alstat** [comment]
Hello Sir,

I got it, there was a problem with the formatting of the data, **sumPokemon.csv** (this is the name of the data in the book) or **pokemonByType.csv**. I found out that the **pokemonByType.csv** has no header name, _type_ and _amount_, which is used in the code, `.value(function(d){return d.amount;})` and `.style("fill", function(d){return color(d.data.type);});`. I compared this with the [pie chart](http://bl.ocks.org/mbostock/3887235) by mbostock, which happens to have a column header, _age_ and _population_, on his csv data.

Thank you for quick response, I am now in Chapter 7. I stop running the codes on each chapter, and just read all the steps. I will get back to programming when I finish this. So far, I have learn a lot on applications of statistical methodologies you presented.

Cheers,

Al
